src/fsio.c

```diff
@@ -2208,7 +2208,6 @@ pr_fs_t *pr_register_fs(pool *p, const char *name, const char *path) {
         name, path);
 
       destroy_pool(fs->fs_pool);
-      fs->fs_pool = NULL;
```

Comment:
The `fs` object here is itself allocated out of the `fs_pool`, thus attempting to dereference the `fs` pointer here, after the call to `destroy_pool()`, is itself a use-after-free case.